### PR TITLE
feat: Forma 36 website preview

### PR DESCRIPTION
**Pull request by bgutsol**
# Purpose of PR

- Add preview logic and api routes

### CODE REVIEW

File: packages/website/pages/api/preview.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  if (secret !== process.env.CONTENTFUL_PREVIEW_ACCESS_TOKEN || !slug) {
+    return res.status(401).json({ message: 'Invalid token' });
+  }
+
+  // Fetch the headless CMS to check if the provided `slug` exists
+  const post = await getSingleArticleBySlug(slug, true);
+
+  // If the slug doesn't exist prevent preview mode from being enabled
+  if (!post) {
+    return res.status(401).json({ message: 'Invalid slug' });
```
Comment (mshaaban0) [inline packages/website/pages/api/preview.ts:15]: Shouldn't this be 404 since it's not found?

**Review by mshaaban0 — APPROVED** [review]
Works very well, thanks

Screenshot 2022-09-07 at 12 54 02

PS: I changed the preview url on Contentful to the preview link of this PR for testing. After merge I assume we will use the default url

File: packages/website/next.config.js [hunk 2]
```diff
@@ -4,5 +4,5 @@ const withMDX = require('@next/mdx')({
 });
 
 module.exports = withMDX({
-  pageExtensions: ['tsx', 'mdx'],
+  pageExtensions: ['ts', 'tsx', 'mdx'],
```
Comment (denkristoffer) [inline packages/website/next.config.js:7]: This PR doesn't add any `.ts` pages, are you sure this is necessary?

File: packages/website/pages/api/preview.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  // Enable Preview Mode by setting the cookies
+  res.setPreviewData({});
+
+  // Redirect to the path from the fetched post
+  // We don't redirect to req.query.slug as that might lead to open redirect vulnerabilities
+  // res.writeHead(307, { Location: `/${post.slug}` })
+  const url = `/${post.slug}`;
+  res.setHeader('Content-Type', 'text/html');
+  res.write(
```
Comment (denkristoffer) [inline packages/website/pages/api/preview.ts:26]: I think you can do something like (pseudo code) `res.status(200).redirect(307, url);` instead.

File: packages/website/next.config.js (hunk 2 above)
Comment (bgutsol) [reply packages/website/next.config.js:7]: yes, files under api folder are `.ts`

Comment (denkristoffer) [reply packages/website/next.config.js:7]: Yeah but they don't render as pages anyway.

Comment (bgutsol) [reply packages/website/next.config.js:7]: NextJs ignores files which extensions are not in this array

**Pull request merged**